apps/admin-portal/src/api/roles.ts

```diff
@@ -54,6 +54,29 @@ export const getGroupsList = (domain: string): Promise<any> => {
         });
 };
 
+/**
+ * Retrieve a list of matched roles according to the search query.
+ * 
+ * @param searchData - search query data
+ */
+export const searchRoleList = (searchData: SearchRoleInterface): Promise<any> => {
+    const requestConfig = {
+        data: searchData,
+        headers: {
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.POST,
+        url: ServiceResourcesEndpoint.groups + "/.search"
+    };
+
+    return httpClient.request(requestConfig).then((response) => {
+        return Promise.resolve(response);
```

Comment:
Maybe we can cast the response to `RoleListInterface`.

```tsx
 response.data as RoleListInterface
```

**Return type**

```
Promise<RoleListInterface>
```

Reply:
Fixed with commit 22dbeb5